src/modules/wc/components/PublicWebComponent.js

```diff
@@ -0,0 +1,583 @@
+			}
+		}
+	}
+
+	/**
+	 * @override
+	 * @protected
+	 */
+	onInitialize() {
+		/**
+		 * Provide the URL for the Iframe considering existing attributes
+		 */
+		const queryParameters = {};
+		for (const attr of this.attributes) {
+			// @ts-ignore
+			if (Object.values(QueryParameters).includes(attr.name)) {
+				this._validateAttributeValue(attr);
+				queryParameters[attr.name] =
+					attr.name === QueryParameters.LAYER
+						? attr.value
+								.split(',')
+								.map((l) => this[l] ?? l)
+								.join()
+						: attr.value;
+			}
+		}
+		this.#iframeUrl = setQueryParams(`${this.#configService.getValueAsPath('FRONTEND_URL')}${PathParameters.EMBED}`, queryParameters);
+
+		/**
+		 * Receive messages from the IFrame
+		 */
+		this.#environmentService.getWindow().parent.addEventListener('message', (event) => this._onReceive(event));
+	}
+
+	/**
+	 * @override
+	 * @protected
+	 */
+	createView() {
+		return html`
+			<style>
+				${css}
+			</style>
+			<iframe
+				src=${this.#iframeUrl}
+				width="100%"
+				height="100%"
+				loading="lazy"
+				frameborder="0"
+				style="border:0"
+				role="application"
+				name=${this.#iFrameId}
+			></iframe>
+		`;
+	}
+
+	_validateAttributeValue(attr) {
+		switch (attr.name) {
+			case QueryParameters.ZOOM:
+				return this.#passOrFail(() => isNumber(attr.value, false), `Attribute "${attr.name}" must be a number`);
+			case QueryParameters.CENTER:
+				return this.#passOrFail(() => !!fromString(attr.value), `Attribute "${attr.name}" must represent a coordinate (easting, northing)`);
+			case QueryParameters.ROTATION:
+				return this.#passOrFail(() => isNumber(attr.value, false), `Attribute "${attr.name}" must be a number`);
+			case QueryParameters.LAYER:
+				return /**No explicit check needed */ true;
+			case QueryParameters.LAYER_VISIBILITY:
+				attr.value
+					.split(',')
+					.forEach((v) => this.#passOrFail(() => isBoolean(v, false), `Attribute "${attr.name}" must contain comma-separated boolean values`));
+				return true;
+			case QueryParameters.LAYER_OPACITY:
+				attr.value
+					.split(',')
+					.forEach((v) =>
+						this.#passOrFail(
+							() => isNumber(v, false) && parseFloat(v) >= 0 && parseFloat(v) <= 1,
+							`Attribute "${attr.name}" must contain comma-separated numbers between 0 and 1`
+						)
+					);
+				return true;
+
+			case QueryParameters.EC_SRID: {
+				const validSRIDs = [4326, this.#mapService.getSrid(), this.#mapService.getLocalProjectedSrid()].map((n) => n.toString());
+				return this.#passOrFail(() => validSRIDs.includes(attr.value), `Attribute "${attr.name}" must be one of [${validSRIDs.join(',')}]`);
+			}
+			case QueryParameters.EC_GEOMETRY_FORMAT: {
+				const validFormats = [SourceTypeName.EWKT, SourceTypeName.GEOJSON, SourceTypeName.KML, SourceTypeName.GPX];
+				return this.#passOrFail(() => validFormats.includes(attr.value), `Attribute "${attr.name}" must be one of [${validFormats.join(',')}]`);
+			}
+		}
+		return false;
```

Comment:
Using the DEFAULT case, opens the opportunity to be expressive (with a additional comment) in a convienient way of the structure of the switch clause too:

```suggestion
        default:
            // we ignore all other attribute candidates
            return false;
			}
		}
		
```